kinds/dashboard/dashboard_kind.cue

```diff
@@ -186,12 +187,160 @@ lineage: schemas: [{
 			state:         #LoadingState
 			error?: {...}
 			description?: string
-			// TODO: Move this into a separated QueryVariableModel type
-			query?:      string | {...}
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variables which allow to select filters from a datasource based on the dimensions from the datasource.
+		#AdHocVariableModel: {
+			#BaseVariableModel
+			type: "adhoc"
+			// Datasource ref can be defined and hold null values
 			datasource?: #DataSourceRef
-			...
-		} @cuetsy(kind="interface") @grafana(TSVeneer="type") @grafanamaturity(NeedsExpertReview)
-		#VariableHide: 0 | 1 | 2                                                 @cuetsy(kind="enum",memberNames="dontHide|hideLabel|hideVariable") @grafana(TSVeneer="type") @grafanamaturity(NeedsExpertReview)
+			filters: [...#AdHocVariableFilter]
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Filters selected filters generated for an ad-hoc variable from a datasource.
+		#AdHocVariableFilter: {
+			key:       string
+			operator:  string
+			value:     string
+			condition: string
+		} @cuetsy(kind="interface")
+
+		// Common props for variables injected by the system.
+		#SystemVariable: {
+			#BaseVariableModel
+			type: "system"
+			current: {
+				value: {...}
+			}
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable injected by the system which holds the current dashboard.
+		#DashSystemVariableModel: {
+			#SystemVariable
+			current: {
+				value: {
+					name: string
+					uid:  string
+				}
+			}
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable injected by the system which holds the current organization.
+		#OrgSystemVariableModel: {
+			#SystemVariable
+			current: {
+				value: {
+					name: string
+					id:   int32
+				}
+			}
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable injected by the system which holds the current user.
+		#UserSystemVariableModel: {
+			#SystemVariable
+			current: {
+				value: {
+					login:  string
+					id:     int32
+					email?: string
+				}
+			}
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variables which value is selected from a list of options.
+		#VariableWithOptions: {
+			#BaseVariableModel
+			current: #VariableOption
```

Comment:
A variable with no options can be an empty object. See https://github.com/grafana/grafana/pull/64358